src/net_processing.cpp

```diff
@@ -1957,6 +1957,19 @@ void PeerManagerImpl::UpdatedBlockTip(const CBlockIndex *pindexNew, const CBlock
     }
 
     m_connman.WakeMessageHandler();
+
+    // Run the ASMap Health check results once per day and log results
+    if (m_connman.UsingASMap() && pindexNew->nHeight % 144 == 0) {
+        const std::vector<CAddress> v4_addrs{m_connman.GetAddresses(0, 0, Network::NET_IPV4)};
+        const std::vector<CAddress> v6_addrs{m_connman.GetAddresses(0, 0, Network::NET_IPV6)};
+        std::vector<CNetAddr> clearnet_addrs;
+        clearnet_addrs.reserve(v4_addrs.size() + v6_addrs.size());
+        std::transform(v4_addrs.begin(), v4_addrs.end(), std::back_inserter(clearnet_addrs),
+            [](const CAddress& addr) { return static_cast<CNetAddr>(addr); });
+        std::transform(v6_addrs.begin(), v6_addrs.end(), std::back_inserter(clearnet_addrs),
+            [](const CAddress& addr) { return static_cast<CNetAddr>(addr); });
+        m_connman.ASMapHealthCheck(clearnet_addrs);
+    }
```

Comment:
I think this belongs in `net.cpp` and shouldn't use the block height.

Reply:
(ignore if you were going to change this once out of draft anyway)

Reply:
I wasn't happy with the location myself but I needed more time to think about where it would fit in best, so happy to take suggestions. I just tried to put it somewhere where it requires the least amount of new dependencies and shipped it to get conceptual feedback.

> and shouldn't use the block height.

What do you suggest to use instead of block height?

Reply:
Sounds good, I am using this now.